skill-tree /src/test/java/com/RDS/skilltree/SkillsIntegrationTests.java

```diff
@@ -1,137 +1,300 @@
+import org.junit.After;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.UUID;
 
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.*;
 
+@Testcontainers
+@Slf4j
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class SkillsIntegrationTests {
+    private UserRepository userRepository;
+    private SkillRepository skillRepository;
+    private static MySQLContainer mysqlContainer;
+    private final UserService userService;
+    private final SkillsService skillsService;
+    private UserDTO user;
+    private SkillDTO skill;
+
+
+    @Autowired
+    public SkillsIntegrationTests(UserService userService, UserRepository userRepository, SkillsService skillsService, SkillRepository skillRepository) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+        this.skillsService = skillsService;
+        this.skillRepository = skillRepository;
+    }
+
+    @LocalServerPort
+    private int port;
+
+    @After
+    private void setup() {
+        RestAssured.baseURI = "http://localhost:";
+    }
+
+    @BeforeEach
+    private void addData() throws MalformedURLException {
+        user = userService.createUser(UserDRO.builder()
+                .role(UserRole.MEMBER)
+                .rdsUserId("p6Bo61VEClhtVdwW0ihg")
+                .lastName("Doe")
+                .firstName("John")
+                .imageUrl(new URL("https://res.cloudinary.com/realdevsquad/image/upload/v1666193594/profile/p6Bo61VEClhtVdwW0iGH/lezguwdq5bgzawa3.jpg"))
+                .build());
 
-    @Before
-    public void setup() {
-        RestAssured.baseURI = "http://localhost:8080";
+        skill = skillsService.createSkill(
+                SkillDRO.builder()
+                        .name("Java")
+                        .type(SkillType.ATOMIC)
+                        .createdBy(user.getId())
+                        .build());
+    }
+
+    @AfterEach
+    private void cleanUp() {
+        skillRepository.deleteAll();
+        userRepository.deleteAll();
+    }
+
+    static {
+        mysqlContainer = new MySQLContainer("mysql:latest")
+                .withDatabaseName("skilltree-test")
+                .withUsername("root")
+                .withPassword("password");
+        mysqlContainer.start();
+    }
+
+    @DynamicPropertySource
```

Comment:
Try using the spring testcontainer dependency
```
<groupId>org.springframework.boot</groupId>
<artifactId>spring-boot-testcontainers</artifactId>
```

This would eliminate the need for having this `@DynamicPropertySource` annotated method.

Instead, you can

1. Create an application.properties under the test/resources
2. Annotate the container with `@ServiceConnection` along with `@Container`

Reply:
Here I was not able to connect to the `test/application.properties`, tried out the TestPropertySource, it didn't work as expected